src/hoc/ClientSideStateUpdate.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import { setAssigneeList } from '@/redux/features/taskBoardSlice'
+import { setTasks, setToken, setWorkflowStates } from '@/redux/features/taskBoardSlice'
+import store from '@/redux/store'
+import { TaskResponse } from '@/types/dto/tasks.dto'
+import { WorkflowStateResponse } from '@/types/dto/workflowStates.dto'
+import { IAssigneeCombined } from '@/types/interfaces'
+import { ReactNode, useEffect } from 'react'
+
+/**
+ * This HOC is responsible in updating the client side state of the responses that are fetched in the server components.
+ * The purpose of this HOC is to avoid prop drilling from server component to client component. Fetched response data from
+ * the server component is used in the client component to achieve the functionalities of the task app.
+ */
+export const ClientSideStateUpdate = ({
```

Comment:
You have the right idea about this store but it requires a little more setup. Please follow this guide: https://redux-toolkit.js.org/usage/nextjs#introduction

Reply:
The redux setup I have done here is already following the doc you've shared above. Can you please let me know if there is anything specific you would like me to do regarding the redux setup? 

Also, this `ClickSideStateUpdate` is not a store but an HOC. It is just responsible for updating the redux store in the client side. This component is a client component and as we can't update redux store from the server component, I am creating this HOC to update the store if necessary. Please let me know your thoughts on this.